app/models/observation.rb

```diff
@@ -345,12 +345,59 @@ def owners_votes
   # Get a list of this User's Votes for this Observation.
   def users_votes(user)
     result = []
-    for n in namings
+    namings.each do |n|
       if v = n.users_vote(user)
         result << v
       end
     end
-    return result
+    result
+  end
+
+  # list of Owners's "favorite" votes for this Observation
```

Comment:
Will remove most comments in this section.  
Please feel free to remove more or to restore any which I removed;  someone who did not write the code is probably better judge of whether the code is self-explanatory.